rlgym_tools/sb3_utils/sb3_log_reward.py

```diff
@@ -1,44 +1,152 @@
+        learner.
+
+        :param reward_functions: Each individual reward function.
+        :param reward_weights: The weights for each reward.
+        """
+        super().__init__(reward_functions,reward_weights)
+        self.index = SB3CombinedLogReward.index_tracker
+        SB3CombinedLogReward.index_tracker += 1
+
+        # make sure there is a folder to dump to
+        os.makedirs('bin/combinedlogfiles', exist_ok=True)
+        self.dumpfile = f'bin/combinedlogfiles/{self.index}.txt'
+
+        # clean out any files that are left over from previous runs
+        if self.index == 0:
+            for file in os.listdir('bin/combinedlogfiles'):
+                if '.txt' in file:
+                    os.remove('bin/combinedlogfiles/'+file)
+
+
+        # initiates the array that will store the rewards
+        self.returns = np.zeros(len(self.reward_functions))
+
+    def reset(self, initial_state: GameState):
+        self.returns = np.zeros(len(self.reward_functions))
+        super().reset(initial_state)
+
+    def get_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        return float(np.dot(self.reward_weights, rewards))
+
+    def get_final_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rewards = [
+            func.get_final_reward(player, state, previous_action)
+            for func in self.reward_functions
+        ]
+
+        self.returns += [a*b for a, b in zip(rewards, self.reward_weights)] # store the rewards
+
+        # write the rewards to file and reset
+        with open(self.dumpfile, 'a') as f:
+            # in a list so we can eval to get it back easily
+            f.write('\n' + str(list(self.returns)))
+        self.returns = np.zeros(len(self.reward_functions))
+
+        return float(np.dot(self.reward_weights, rewards))
+
+
+class SB3CombinedLogRewardCallback(BaseCallback):
+    def __init__(self, rew_names:List[str]):
+        """
+        Callback to log the data from a SB3CombinedLogReward to the
+        same log as the model.
+
+        :param rew_names: List of names that the logger will use for
+        each reward.
+        """
+        super().__init__()
+        self.rew_names = rew_names
+
+    def _on_step(self) -> bool:
+        return True
+
+    def _on_rollout_end(self) -> None:
+        returns = []
+        # get all returns out of the files
+        for file in os.listdir('bin/combinedlogfiles'):
+            if '.txt' in file:
+                with open('bin/combinedlogfiles/'+file, 'r') as f:
+                    for line in f:
+                        if line != '\n':
+                            returns.append(eval(line.strip()))

```

Comment:
unnecessary eval.
Also should really be wrapped in a try-except block given that this part is likely to error.

Reply:
Overall I think the writing above should be changed to a json.dumps(), in which case this should be a json.loads().